src/context.cpp

```diff
@@ -483,6 +484,102 @@ namespace librealsense
         query_option_callback _query_opt_cb;
     };
 
+
+    // Frame data and metadata are sent as two seperate streams.
+    // They are synchronized using frame_id for matching data+metadata sets.
+    template< class T >
+    class frame_metadata_syncer
+    {
+    public:
+        // We don't want the queue to get large, it means lots of drops and data that we store to (probably) throw later
+        static constexpr size_t max_md_queue_size = 8;
+        // If a metadata is lost we wait for it until the next frame arrives, causing a small delay but we prefer passing
+        // the frame late and without metadata over losing it.
+        static constexpr size_t max_frame_queue_size = 2;
+
+        void enqueue_frame( T && frame, std::shared_ptr< rs2_stream_profile > prof )
+        {
+            std::lock_guard< std::mutex > lock( _queues_lock );
+            _frame_queue.push_back( std::move( frame ) );
+            _prof_queue.push_back( prof );
```

Comment:
Is this used?

Reply:
Yes. Due to awkward `rs2_software_video_frame` I had to keep a `rs_stream_profile` pointer for the life time of the frame in the queue (until `on_video_frame` is called)

Reply:
So please add a comment -- I saw no usage and it's not explained.

Reply:
There is a comment when calling this [here](https://github.com/IntelRealSense/librealsense/pull/11412/files/5d454bde748ad56ee2dff3fb22782b4e6e50b4c9#diff-776a5f8330d9fb807e058595e98557798a24587a21f080bb5d77542d50875acfR713), but you are right, I will add one here too